issue when Adding Cloze Note On some android phones

https://1drv.ms/v/s!Aoq6hTbE2xg2jBFWJ44iBnAmTbbD

That link contains the screen recording of how the issue took place. Also if you need logs, please tell how to do them. This issue is persistent with many other friends who use ankidroid, making us have to use the computer app in order to create cloze notes.Other stuff:

If you need any more information, just reply i will reply within the same day. For logs tell me how to do them plz. That exact thing happens on my 2 devices:
Oneplus 8, Xiaomi K20 Pro. The app version is:2.15.6

No crash reports with that UUID unfortunately.

Got this from a K20 though

```


Stack Trace | com.ichi2.utils.JSONException:  org.json.JSONException: Value null at did of type org.json.JSONObject$1  cannot be converted to long	at com.ichi2.utils.JSONObject.getLong(JSONObject.java:2)	at com.ichi2.anki.NoteEditor$SetNoteTypeListener.onItemSelected(NoteEditor.java:12)	at android.widget.AdapterView.fireOnSelected(AdapterView.java:957)	at android.widget.AdapterView.dispatchOnItemSelected(AdapterView.java:946)	at android.widget.AdapterView.access$300(AdapterView.java:55)	at android.widget.AdapterView$SelectionNotifier.run(AdapterView.java:910)	at android.os.Handler.handleCallback(Handler.java:938)	at android.os.Handler.dispatchMessage(Handler.java:99)	at android.os.Looper.loop(Looper.java:223)	at android.app.ActivityThread.main(ActivityThread.java:7665)	at java.lang.reflect.Method.invoke(Native Method)	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:594)	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)Caused by: org.json.JSONException: Value null at did of type org.json.JSONObject$1 cannot be converted to long	at org.json.JSON.typeMismatch(JSON.java:101)	at org.json.JSONObject.getLong(JSONObject.java:528)	at com.ichi2.utils.JSONObject.getLong(JSONObject.java:1)	... 12 more
-- | --



```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
@@ -53,12 +53,10 @@ import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
-import android.view.ViewGroup;
 import android.view.ViewGroup.MarginLayoutParams;
 import android.view.WindowManager;
 import android.widget.AdapterView;
 import android.widget.AdapterView.OnItemSelectedListener;
-import android.widget.ArrayAdapter;
 import android.widget.EditText;
 import android.widget.ImageButton;
 import android.widget.LinearLayout;
@@ -98,6 +96,7 @@ import com.ichi2.async.TaskManager;
 import com.ichi2.compat.CompatHelper;
 import com.ichi2.libanki.Card;
 import com.ichi2.libanki.Collection;
+import com.ichi2.libanki.Consts;
 import com.ichi2.libanki.Models;
 import com.ichi2.libanki.Model;
 import com.ichi2.libanki.Note;
@@ -112,7 +111,6 @@ import com.ichi2.utils.FunctionalInterfaces.Consumer;
 import com.ichi2.utils.HashUtil;
 import com.ichi2.utils.KeyUtils;
 import com.ichi2.utils.MapUtil;
-import com.ichi2.utils.NamedJSONComparator;
 import com.ichi2.utils.NoteFieldDecorator;
 import com.ichi2.utils.TextViewUtil;
 import com.ichi2.widget.WidgetStatus;
@@ -122,7 +120,6 @@ import com.ichi2.utils.JSONObject;
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.LinkedList;
 import java.util.List;
@@ -2081,7 +2078,7 @@ public class NoteEditor extends AnkiActivity implements
                 getCol().getDecks().save(currentDeck);
                 // Update deck
                 if (!getCol().get_config("addToCur", true)) {
-                    mCurrentDid = model.getLong("did");
+                    mCurrentDid = model.optLong("did", Consts.DEFAULT_DECK_ID);
                 }
 
                 refreshNoteData(FieldChangeType.changeFieldCount(shouldReplaceNewlines()));
@@ -2308,6 +2305,15 @@ public class NoteEditor extends AnkiActivity implements
         return mEditFields.get(index);
     }
 
+
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    void setCurrentlySelectedModel(long mid) {
+        int position = mAllModelIds.indexOf(mid);
+        if (position == -1) {
+            throw new IllegalStateException(mid + " not found");
+        }
+        mNoteTypeSpinner.setSelection(position);
+    }
 
     private class EditFieldTextWatcher implements TextWatcher {
         private final int mIndex;

--- a/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java
@@ -39,6 +39,7 @@ import org.robolectric.shadows.ShadowActivity;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 import androidx.test.core.app.ActivityScenario;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
@@ -343,6 +344,20 @@ public class NoteEditorTest extends RobolectricTest {
         NoteEditor editor = getNoteEditorAddingNote(FromScreen.DECK_LIST, NoteEditor.class);
 
         assertThat("Fields should have their first word capitalized by default", editor.getFieldForTest(0).isCapitalized(), is(true));
+    }
+
+    @Test
+    @Config(qualifiers = "en")
+    public void addToCurrentWithNoDeckSelectsDefault_issue_9616() {
+        getCol().getConf().put("addToCur", false);
+        Model cloze = Objects.requireNonNull(getCol().getModels().byName("Cloze"));
+        cloze.remove("did");
+        getCol().getModels().save(cloze);
+        NoteEditor editor = getNoteEditorAddingNote(FromScreen.DECK_LIST, NoteEditor.class);
+
+        editor.setCurrentlySelectedModel(cloze.getLong("id"));
+
+        assertThat(editor.getDeckId(), is(Consts.DEFAULT_DECK_ID));
     }
 
     private Intent getCopyNoteIntent(NoteEditor editor) {

